client/my-sites/customer-home/controller.jsx

```diff
@@ -29,21 +29,21 @@ export function maybeRedirect( context, next ) {
 	}
 
 	const siteId = getSelectedSiteId( state );
-	const options = getSiteOptions( state, siteId );
+	const maybeStalelaunchpadScreenOption = getSiteOptions( state, siteId )?.launchpad_screen;
 
-	// Normally, checking the launchpad_screen option in redux state would be enough to decide whether
-	// or not to redirect to launchpad. The option, however, is loading stale data in horizon, and presumably,
-	// in production as well. The forceLoadLaunchpadData query param is a temporary patch to circumvent stale data, and
-	// will avoid a redirect.
-	const shouldRedirectToLaunchpad =
-		options?.launchpad_screen === 'full' &&
-		! getQueryArg( window.location.href, 'forceLoadLaunchpadData' );
+	if ( maybeStalelaunchpadScreenOption && maybeStalelaunchpadScreenOption === 'full' ) {
+		await context.store.dispatch( requestSite( siteId ) );
+	}
+
+	// TODO: Write explanation for refetching launchpad site options
```

Comment:
I am wondering why we fetch the same data twice? I assume it's because we want to avoid the dispatch call on line 35 unless we need it. Also we think the data may change after that dispatch call. But that raises a secondary question of why whether it makes sense to base the conditional on line 32 on stale data? I assume we think this is ok because it is only likely to be stale in a safe direction (ie, stale data says full even though launchpad is really off, rather than stale data says off when launchpad is really full).

Reply:
> I am wondering why we fetch the same data twice? I assume it's because we want to avoid the dispatch call on line 35 unless we need it

Correct

> Also we think the data may change after that dispatch call. But that raises a secondary question of why whether it makes sense to base the conditional on line 32 on stale data?

That was discussed earlier here https://github.com/Automattic/wp-calypso/pull/67967#discussion_r974590042